### DISCUSSION TITLE
Inconsistency with bumper operation

### DISCUSSION THREAD

**Discussion by iansexton1** [iRobotEducation/create3_docs#169] [Questions] [answered]
I wrote a sequence for the robot to backup and turn 45° when the bumper is pressed.
```
 @event(robot.when_bumped, [])
    async def bumped(robot):
    await robot.set_lights_rgb(255, 0, 0)
    await backoff(robot)
    await forward(robot)

async def backoff(robot):
    await robot.move(-30)
    await robot.turn_left(-45)
```
The issue I'm having is that when the robot bumps into a couch its not backing up at all but just turns the 45° but when it hits other objects such as a box it performs as intended. What is it about when it hits the couch that it doesnt backup before turning 45°

https://user-images.githubusercontent.com/109889851/185780282-a2e93ce0-390f-4636-a745-72c137a7f718.MOV

https://user-images.githubusercontent.com/109889851/185780315-bd30ab02-a288-48da-8329-3f1cc6aae2ee.MOV

**Comment by maddiepero** [comment]
Hi! In the second video, the robot appears to slow down before approaching the box and the light color changes to orange, however, it doesn't do that in the first video. Are you using the IR sensors to trigger that? The couch obviously doesn't make the code act any differently so I am wondering if it's either getting stuck in a different part of the code because the IR sensors aren't picking up an object or if the couch isn't completely pressing the bumper since it's a softer material than the box. How much the bumper is pressed by the couch is hard to tell from the video.

**Reply by alsora** [reply]
Note that robots have a "proximity slowdown" reflex that is automatically enabled and indeed uses IR sensors to detect incoming obstacles.
This can be configured through the ROS 2 parameters of the motion control node https://iroboteducation.github.io/create3_docs/api/reflexes/

**Reply by iansexton1** [reply]
The slowing down in the second video was done intentionally, we wrote the following code to hopefully remedy the issue with the couch but it appears that because of the gap in the couch it doesn't get picked up by the IR sensors
while True:
        sensors = (await robot.get_ir_proximity()).sensors
        if front_obstacle0(sensors):
            await slowdown(robot)
        if front_obstacle1(sensors):
            await slowdown(robot)
        if front_obstacle2(sensors):
            await slowdown(robot)
etc...
async def slowdown(robot):
    await robot.set_lights_rgb(255, 80, 0)
    await robot.set_wheel_speeds(speed1, speed1)  speed1=3

**Comment by shamlian** [comment, marked as answer]
This is caused by the ROS 2 reflexes interfering with the iRobot Coding commands. In a subsequent update, we will disable all reflexes when the iRobot Coding mode is active. In the mean time, you could disable reflexes via the motion control node as @alsora described above.

**Reply by shamlian** [reply]
I _believe_ the easiest way to do this from the web interface is to add `reflexes_enabled: false` to the end of the [Application ROS 2 Parameters File](https://iroboteducation.github.io/create3_docs/webserver/application/#application-configuration).

**Comment by halimimaha** [comment]
@shamlian i want to disable robot's bumper so i tried to  add reflexes.REFLEX_BUMP: false in the end of Application Configuration but still enable :/ do u have any suggestions  : 
motion_control:
ros__parameters:
safety_override: "none"
reflexes.REFLEX_BUMP: false

**Reply by alsora** [reply]
@halimimaha, please don't cross-post the same questions in multiple places.
I replied you in your first post https://github.com/iRobotEducation/create3_docs/issues/533#issuecomment-1919329202.
Let's keep using that one
